apps/smart-contracts/core/test/TokenSender.test.ts

```diff
@@ -225,4 +231,103 @@ describe('=> TokenSender', () => {
       )
     })
   })
+
+  describe('# send', () => {
+    beforeEach(async () => {
+      await tokenSender.connect(deployer).setPrice(priceOracle.address)
+      await tokenSender.connect(deployer).setAllowedCallers([deployer.address], [true])
+    })
+
+    async function configureValues(price: number, lowerBound: number): Promise<void> {
+      priceOracle.get.returns(price)
+      await tokenSender.connect(deployer).setPriceMultiplier(MULTIPLIER_DENOMINATOR)
+      await tokenSender.connect(deployer).setScaledPriceLowerBound(lowerBound)
+    }
+
+    async function getOutputAmount(unconvertedAmount: number): Promise<BigNumber> {
+      const scaledPrice = (await priceOracle.get())
+        .mul(await tokenSender.getPriceMultiplier())
+        .div(MULTIPLIER_DENOMINATOR)
+      expect(scaledPrice).to.be.gt(await tokenSender.getScaledPriceLowerBound())
+      const outputTokenDecimalsFactor = ethers.BigNumber.from(10).pow(await outputToken.decimals())
+      const outputAmount = ethers.BigNumber.from(unconvertedAmount)
+        .mul(outputTokenDecimalsFactor)
+        .div(scaledPrice)
+      return outputAmount
+    }
+
+    it('reverts if not allowed caller', async () => {
+      expect(await tokenSender.isCallerAllowed(user.address)).to.eq(false)
+
+      await expect(tokenSender.connect(user).send(user.address, 1)).revertedWith(
+        `msg.sender not allowed`
+      )
+    })
+
+    it("doesn't transfer if unconvertedAmount = 0", async () => {
+      await configureValues(2, 1)
+
+      await tokenSender.connect(deployer).send(user.address, 0)
+
+      expect(outputToken.transfer).to.not.be.called
+    })
+
+    it("doesn't transfer if price = 0 and priceMultiplier > 0", async () => {
```

Comment:
```suggestion
    it("doesn't transfer if price = 0", async () => {
```
(doesn't matter what the other variables are when price = 0, with unconvertedAmount > 0 being implied given above test)